### PR TITLE
KAZOO-5117: convert body to binary to make sure content_lenght sets correctly

### PR DESCRIPTION
**Pull request by icehess**
and some code cleanup

### CODE REVIEW

File: core/kazoo_web/src/kz_http.erl [hunk 1]
```diff
@@ -279,7 +279,11 @@ build_request(Method, Url, Headers, Body) when (Method == 'post');
                                           ]
                                          ,Headers
                                          ,""),
-    {kz_util:to_list(Url), ensure_string_headers(Headers), kz_util:to_list(ContentType), Body}.
+    {kz_util:to_list(Url)
+    ,ensure_string_headers(Headers)
+    ,kz_util:to_list(ContentType)
+    ,kz_util:to_binary(Body)
```
Comment (fenollp) [inline core/kazoo_web/src/kz_http.erl]: How about using `iolist_to_binary` instead?
Also, how about fixing all the calls to kz_http:req calls that call to_binary(Body)? This should save some time/confusion.

**Comment by fenollp** [comment]
@lazedo @icehess I believe the comment on removing superfluous calls to to_binary/1 in code that uses kz_http is still relevant ;)

**Comment by lazedo** [comment]
@fenollp agree. open a jira ticket and work on it :)

**Comment by icehess** [comment]
@fenollp @lazedo Why? most of the time they call `kz_json:encode` or `unicode:*_to_binary` on Body which is not related to kz_http.  kz_http accepts both string(or iolist) and binary and at the end for consistency convert it to binary.

**Pull request merged**